test/e2e/common.go

```diff
@@ -63,6 +113,41 @@ func DeleteTestNamespace(t *testing.T, namespace string) {
 	}
 }
 
+// WaitForNamespaceDeleted wait until namespace is deleted
+func WaitForNamespaceDeleted(t *testing.T, namespace string) {
+	logger := Logger{}
+	kubectlGetNamespace := func() string {
+		kubectl := kubectl{t, logger}
+		out, err := kubectl.RunWithOpts([]string{"get", "namespace"}, runOpts{})
+		if err != nil {
+			t.Fatalf(fmt.Sprintf("Error executing 'kubectl get namespace' command. Error: %s", err.Error()))
+			return ""
+		}
+
+		return out
+	}
+
+	logger.Debugf("Waiting for namespace: %s to be deleted", namespace)
+	deleted := false
```

Comment:
as above, if you put the following code in a function you could return with a nil error on success and avoid the usage of such flags (which I consider bad style as its hard to follow and are often unnecessary).

This would make the code much more readable.

Reply:
Again slightly more complicated. But see update that separates retry logic into private func.